### ISSUE TITLE
application gets stuck 

### ISSUE THREAD

**Issue by jaywakeford** [stackblitz/bolt.new#907]
### Describe the bug

One example is :

'what should I do next can I run the sql query directly in pgadmin

prompt is too long: 200052 tokens > 199999 maximum'  

A single sentence should never cause this issue.  

I also frequently, intermittent but frequently get a '
This message didn't complete its response.'  error.   

Once it gets it burns tokens and take many iterations to get passed the issue.   Overnight waits, restarts, refresh browser are all fruitless as solutions.  

Estimating that 35% of token burn gets used in these two scenarios.   Plus lost time.   

### Link to the Bolt URL that caused the error

https://bolt.new/~/sb1-wlgkaw

### Steps to reproduce

Intermittent and seemingly in illogical times.  ie above where 1 line of a message results in too many tokens for the prompt... 

### Expected behavior

If stuck the resolution process should be smoother at worst and at best not require more than 2 iterations to resolve.  I have experienced 10-15 iterations before resolution.  

### Platform

```
Browser name = Chrome
Full version = 130.0.0.0
Major version = 130
navigator.appName = Netscape
navigator.userAgent = Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Safari/537.36
performance.memory = {
  "totalJSHeapSize": 94460588,
  "usedJSHeapSize": 91675216,
  "jsHeapSizeLimit": 4294705152
}
Username = jaywakeford
Chat ID = 7750a7f61771
```

**Comment by kc0tlh** [comment]
@jaywakeford thanks for your report, and appreciate your patience as we are a small team working to support all of the new users! The context on this error, workaround ideas, and future updates on the R&D we are doing on this issue are being tracked in #1322 going forward so please go subscribe there!

**Comment by jaywakeford** [comment]
Hi Alex ,  I took a few days break from burning more tokens while the team worked through the issues.  Today I went in and requested the directory structure of my project in MD and it was successful with no pushback yet. 🏆  I will tentatively continue building.   Will report any issues as they arrive.   Happy to be building again , albeit my pocket book is not.    I hope the team is diligently working on a strategy that ensures pricing remains competitive.  Warm regards, Jay WakefordOn 1 Nov 2024, at 3:15 PM, Alexander Berger ***@***.***> wrote:﻿
Closed #907 as completed.

**Comment by kc0tlh** [comment]
@jaywakeford we are doing everything we can in #678 to ensure we bring cost-per-message down while increasing reliability at the same time! I appreciate your patience as one of our earliest users, and hope we can improve the app fast enough to retain you for life!

**Comment by jaywakeford** [comment]
#1407 describes the issue that did arise again shortly after I started using it again.  Reluctant to send all the issues as it is just on repeat.  Not sure what else to say at this point.  Heading to the open source release and it’s messing up my other 4 projects significantly while waiting but I trust your team gets it and has some moral compass to correct the issues quickly for the sake of your initial supporters and paying clients.   Warm regards,  Jay WakefordOn 2 Nov 2024, at 1:48 PM, Jay Wakeford ***@***.***> wrote:﻿Hi Alex ,  I took a few days break from burning more tokens while the team worked through the issues.  Today I went in and requested the directory structure of my project in MD and it was successful with no pushback yet. 🏆  I will tentatively continue building.   Will report any issues as they arrive.   Happy to be building again , albeit my pocket book is not.    I hope the team is diligently working on a strategy that ensures pricing remains competitive.  Warm regards, Jay WakefordOn 1 Nov 2024, at 3:15 PM, Alexander Berger ***@***.***> wrote:﻿
Closed #907 as completed.
